.palantir/revapi.yml

```diff
@@ -3,12 +3,19 @@ versionOverrides:
 acceptedBreaks:
   apache-iceberg-0.14.0:
     org.apache.iceberg:iceberg-api:
+    - code: "java.class.defaultSerializationChanged"
```

Comment:
Why is partition key complaining here?  Can we not add this one?

Reply:
I don't understand the reason why the partition key is complaining either. 
When I try to execute ` ./gradlew :iceberg-api:revapi --rerun-tasks`

I get the following error.

```
Execution failed for task ':iceberg-api:revapi'.
> There were Java public API/ABI breaks reported by revapi:
  
  java.class.defaultSerializationChanged: The default serialization ID for the class has changed. This means that the new version of the class is not deserializable from the byte stream of a serialized old class.
  
  old: class org.apache.iceberg.PartitionKey
  new: class org.apache.iceberg.PartitionKey
  
  SOURCE: EQUIVALENT, BINARY: EQUIVALENT, SEMANTIC: BREAKING
  
  From old archive: iceberg-api-0.14.0.jar
  From new archive: iceberg-api-0.15.0-SNAPSHOT.jar
```

Reply:
I didn't quite understand why this was needed. However, this was already added upstream, before this PR got merged. So, I think we can resolve this conversation. I updated the PR to fix the merge conflict.